lptypes.pas

```diff
@@ -583,7 +583,9 @@   TLapeManagingDeclaration = class(TLapeDeclaration)
 
   LabelOperators = CompareOperators;
   EnumOperators = [op_Plus, op_Minus, op_Assign] + CompareOperators;
-
+  OverloadableOperators = [op_UnaryMinus, op_NOT, op_Assign, op_Plus, op_Minus, op_Multiply, op_Divide, op_DIV, op_Power, op_MOD, op_IN, op_SHL, op_SHR] + CompareOperators + BinaryOperators;
+  
+  
   op_str: array[EOperator] of lpString = ('',
     '=', '>', '>=', '<', '<=', '<>', '@', 'and', ':=', '^', 'div', '/', '.' , 'in',
     '[', '-', 'mod', '*', 'not', 'or', '+', '**', 'shl', 'shr', 'xor', '-', '+');
```

Comment:
In the current setup, it would only allow for read-only index operations, but I guess disabling it is also an option.